### PR TITLE
Add react routing to app

### LINKED ISSUE #48: Add routing module (3)
To support the site having multiple pages, we need to add React Router to our application.

### PR DESCRIPTION
**Pull request by spanjshiri**
This pr does a slight refactor of the front end application to use react router and nests components within the main app component which is now responsible for routing. 

This relates to issue #48

### CODE REVIEW

File: ui/src/Upload.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useReducer } from 'react';
+import { AppDispatchContext, AppDispatchReducer, initialState } from './reducers/AppDispatchReducer';
+import './Upload.css';
+import MemeUpload from './MemeUpload';
+import ImageUpload from './ImageUpload';
+
+function Upload(){
```
Comment (KatherynRizek) [inline ui/src/Upload.tsx]: Should the name of this be more like `CreateMeme` rather than `Upload`? Curious on thoughts

**Review by KatherynRizek — DISMISSED** [review]

File: ui/src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
-        <h1 className="inputPrompt">Create Your Meme</h1>
-        <div className="memeCreation">
-          <AppDispatchContext.Provider value={value}>
-            <ImageUpload topText={appState.topText} bottomText={appState.bottomText} memeContentRef={appState.memeContentRef}/>
-            <MemeUpload memeName={appState.memeName} imageID={appState.imageID} memeContentRef={appState.memeContentRef} memeResultRef={appState.memeResultRef}/>
-          </AppDispatchContext.Provider>
-        </div>
-      </header>
-    </div>
+    <main>
```
Comment (mchughmk) [inline ui/src/App.tsx]: Replace `main` with `BrowserRouter` here.

File: ui/src/index.tsx [hunk 3]
```diff
@@ -1,10 +1,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { BrowserRouter} from "react-router-dom";
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(<BrowserRouter>
```
Comment (mchughmk) [inline ui/src/index.tsx]: `BrowserRouter` should be inside `App`, as noted above.  You should only render `App` here.

**Review by mchughmk — CHANGES_REQUESTED** [review]

**Review by KatherynRizek — DISMISSED** [review]

File: ui/src/App.tsx [hunk 4]
```diff
@@ -1,25 +1,17 @@
-import React, { useReducer } from 'react';
-import { AppDispatchContext, AppDispatchReducer, initialState } from './reducers/AppDispatchReducer';
-import './App.css';
-import MemeUpload from './MemeUpload';
-import ImageUpload from './ImageUpload';
+import React from 'react';
+import {Route, Switch, BrowserRouter } from "react-router-dom";
```
Comment (mchughmk) [inline ui/src/App.tsx]: Convention is to import `BrowserRouter as Router`, then just render `Router`.

**Review by mchughmk — CHANGES_REQUESTED** [review]

File: ui/src/App.tsx (hunk 4 above)
Comment (spanjshiri) [reply ui/src/App.tsx]: ok sad face. I will fix it now

**Review by mchughmk — APPROVED** [review]

**Review by KatherynRizek — APPROVED** [review]

**Pull request merged**